types/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+pub use crate::data_types::{Column, DataTypeNode};
+use crate::decoders::{ensure_size, read_string};
+use crate::error::TypesError;
+pub use crate::leb128::put_leb128;
+pub use crate::leb128::read_leb128;
```

Comment:
A leaky abstraction. We export these functions only because we want to reuse them, but logically they aren't part of the public API (because it's an implementation detail) and can be moved out in the future.

Let's mark them "doc(hidden)".